frontend/src/metabase/admin/tasks/components/LogLevelsModal/DurationInput.tsx

```diff
@@ -0,0 +1,62 @@
+import { t } from "ttag";
+
+import { Flex, Select, TextInput } from "metabase/ui";
+import type { TimeUnit } from "metabase-types/api";
+
+interface Props {
+  duration: string;
+  durationUnit: TimeUnit;
+  onDurationChange: (duration: string) => void;
+  onDurationUnitChange: (durationUnit: TimeUnit) => void;
+}
+
+export const DurationInput = ({
+  duration,
+  durationUnit,
+  onDurationChange,
+  onDurationUnitChange,
+}: Props) => {
+  return (
+    <Flex align="flex-end" gap="md">
+      <TextInput
+        label={t`Duration`}
+        placeholder={t`Duration`}
+        required
+        type="number"
+        value={duration}
+        w={80}
+        onChange={(event) => onDurationChange(event.target.value)}
+      />
+
+      <Select
+        data={getData()}
+        placeholder={t`Unit`}
+        value={durationUnit}
+        w={140}
+        onChange={onDurationUnitChange}
+      />
+    </Flex>
+  );
+};
+
+// Some options are not practically useful
+type AllowedTimeUnit = Exclude<
+  TimeUnit,
+  "nanoseconds" | "microseconds" | "milliseconds"
+>;
```

Comment:
(mega-)nit: `Pick` would be the more conservative/defensive option here, to avoid other options from unexpectedly showing up in the future.

Reply:
> to avoid other options from unexpectedly showing up in the future.

`getData` function below already has protection against this.

With `Pick` we wouldn't get an error when `LoggerDurationUnit` is extended.